src/lib/app-provider/hooks/useBaseApiRoute.ts

```diff
@@ -0,0 +1,27 @@
+import { useCelatoneApp } from "../contexts";
+
+export const useBaseApiRoute = (
+  type: "txs" | "balances" | "assets" | "projects" | "contracts" | "codes"
+): string => {
+  const {
+    chainConfig: { chain, api },
+    currentChainId,
+  } = useCelatoneApp();
+  if (!chain || !api || !currentChainId) return "";
```

Comment:
I believe we should throw an error if any of these values is undefined or an empty string.